### PR TITLE
Update zksync crate to master branch

### PR DESCRIPTION
**Pull request by maaktweluit**
also updated num to 0.3 and bigdecimal to 0.2

### CODE REVIEW

File: core/payment-driver/zksync/src/zksync/utils.rs [hunk 1]
```diff
@@ -45,9 +44,9 @@ pub fn pack_up(amount: &BigUint) -> BigUint {
 
 fn increase_least_significant_digit(amount: &BigUint) -> BigUint {
     let digits = amount.to_radix_le(10);
-    for i in 0..(digits.len()) {
+    for i in 0..digits.len() {
         if digits[i] != 0 {
-            return amount + pow(BigUint::from(10u32), i);
+            return amount + BigUint::from(10u32).pow(i.try_into().unwrap());
```
Comment (maaktweluit) [inline core/payment-driver/zksync/src/zksync/utils.rs:49]: `pow` was the only reason why `num` was imported over `num-bigint`

any idea's how to remove this try_into? usize -> u32 should always be safe AFAIK

File: Cargo.toml [hunk 2]
```diff
@@ -172,11 +172,11 @@ ya-sb-router = { git = "https://github.com/golemfactory/ya-service-bus.git", rev
...
 
 ## CLIENT
-ya-client = { git = "https://github.com/golemfactory/ya-client.git", rev = "4099adb99dadbc8929caba33538806f5e10d5920"}
-ya-client-model = { git = "https://github.com/golemfactory/ya-client.git", rev = "4099adb99dadbc8929caba33538806f5e10d5920"}
+# ya-client = { git = "https://github.com/golemfactory/ya-client.git", rev = "4099adb99dadbc8929caba33538806f5e10d5920"}
+# ya-client-model = { git = "https://github.com/golemfactory/ya-client.git", rev = "4099adb99dadbc8929caba33538806f5e10d5920"}
 
-#ya-client = { path = "../ya-client" }
-#ya-client-model = { path = "../ya-client/model" }
+ya-client = { path = "../ya-client" }
```
Comment (maaktweluit) [inline Cargo.toml:178]: TODO: update to `payments/beta-1` branch after ya-client is merged

**Review by nieznanysprawiciel — APPROVED** [review]

File: core/payment-driver/zksync/src/zksync/utils.rs (hunk 1 above)
Comment (Wiezzel) [reply core/payment-driver/zksync/src/zksync/utils.rs:49]: Just use `i as u32`

File: core/payment-driver/zksync/Cargo.toml [hunk 3]
```diff
@@ -11,21 +11,21 @@ default = []
...
+num-bigint = { version = "0.3", features = ["serde"] }
 serde = "1.0"
 serde_json = "^1.0"
 tiny-keccak = "1.4.2"
 tokio = { version = "0.2", features = ["full"] }
 uuid = { version = "0.8", features = ["v4"] }
-zksync = { git = "https://github.com/matter-labs/zksync", rev = "9b4263c883c0f57c75068a0cc6339d051157a306"}
-zksync_eth_signer = { git = "https://github.com/matter-labs/zksync", rev = "9b4263c883c0f57c75068a0cc6339d051157a306"}
+zksync = { git = "https://github.com/matter-labs/zksync", branch = "master"}
+zksync_eth_signer = { git = "https://github.com/matter-labs/zksync", branch = "master"}
```
Comment (Wiezzel) [inline core/payment-driver/zksync/Cargo.toml:28]: Shouldn't we rather fix the version to a particular commit?

File: core/payment-driver/zksync/src/zksync/wallet.rs [hunk 4]
```diff
@@ -267,7 +267,7 @@ async fn unlock_wallet<S: EthereumSigner + Clone>(
             .send()
             .await
             .map_err(|e| GenericError::new(format!("Failed to send change_pubkey request: '{}'. HINT: Did you run `yagna payment fund` and follow the instructions?", e)))?;
-        log::debug!("Unlock tx: {:?}", unlock);
+        // DO WE NEED DEBUG? log::debug!("Unlock tx: {:?}", unlock);
```
Comment (Wiezzel) [inline core/payment-driver/zksync/src/zksync/wallet.rs:270]: I'd suggest `log::trace()` for this.

File: core/payment-driver/zksync/src/zksync/wallet.rs [hunk 5]
```diff
@@ -333,6 +333,6 @@ pub async fn withdraw<S: EthereumSigner + Clone>(
         .await
         .map_err(GenericError::new)?;
 
-    debug!("Withdraw handle: {:?}", withdraw_handle);
+    // TODO: Do we need this handle debugged? debug!("Withdraw handle: {:?}", withdraw_handle);
```
Comment (Wiezzel) [inline core/payment-driver/zksync/src/zksync/wallet.rs:336]: Not really.

File: core/payment-driver/zksync/src/zksync/utils.rs (hunk 1 above)
Comment (maaktweluit) [reply core/payment-driver/zksync/src/zksync/utils.rs]: that works, thanks :D

File: core/payment-driver/zksync/Cargo.toml (hunk 3 above)
Comment (maaktweluit) [reply core/payment-driver/zksync/Cargo.toml]: yes, done :)

File: core/payment-driver/zksync/src/zksync/wallet.rs (hunk 4 above)
Comment (maaktweluit) [reply core/payment-driver/zksync/src/zksync/wallet.rs:289]: made it a cleaner message, so kept it as debug.. please check!

File: core/payment-driver/zksync/src/zksync/wallet.rs (hunk 5 above)
Comment (maaktweluit) [reply core/payment-driver/zksync/src/zksync/wallet.rs]: made it a cleaner message and kept it, please check

File: agent/provider/src/payments/pricing.rs [hunk 6]
```diff
@@ -52,7 +52,8 @@ impl PaymentModel for LinearPricing {
                 .zip(usage.iter())
                 .map(|(coeff, usage_value)| coeff * usage_value)
                 .sum::<f64>();
-        Ok(BigDecimal::from(cost))
+
+        BigDecimal::from_f64(cost).ok_or(anyhow!("Failed to convert to BigDecimal: {}", cost))
```
Comment (Wiezzel) [inline agent/provider/src/payments/pricing.rs:56]: I don't understand this change. Why use `from_f64` if you can use `from` and skip the unnecessary error handling? This operation should always succeed.

**Review by Wiezzel — APPROVED** [review]

File: agent/provider/src/payments/pricing.rs (hunk 6 above)
Comment (Wiezzel) [reply agent/provider/src/payments/pricing.rs:56]: Ah, I see, they removed it.

File: core/payment-driver/zksync/src/zksync/wallet.rs [hunk 7]
```diff
@@ -333,16 +342,19 @@ pub async fn withdraw<S: EthereumSigner + Clone>(
...
-        .amount(withdraw_amount)
-        .to(recipient_address)
-        .send()
-        .await
-        .map_err(GenericError::new)?;
+        .amount(withdraw_amount.clone())
+        .to(recipient_address);
+    log::debug!(
+        "Withdrawal raw data. token={}, amount={}, to={}",
+        ZKSYNC_TOKEN_NAME,
```
Comment (tworec) [inline core/payment-driver/zksync/src/zksync/wallet.rs:353]: you have tGLM hardcoded above in `info!`

File: core/payment-driver/zksync/src/zksync/wallet.rs (hunk 4 above)
Comment (tworec) [reply core/payment-driver/zksync/src/zksync/wallet.rs:289]: seems trash. Pls resolve it

File: Cargo.toml [hunk 8]
```diff
@@ -174,11 +174,11 @@ ya-sb-router = { git = "https://github.com/golemfactory/ya-service-bus.git", rev
 ya-sb-util = { git = "https://github.com/golemfactory/ya-service-bus.git", rev = "6b494e17d7a662e0b710af8c5a2e99ab4007fdb9"}
 
 ## CLIENT
-ya-client = { git = "https://github.com/golemfactory/ya-client.git", rev = "4099adb99dadbc8929caba33538806f5e10d5920"}
-ya-client-model = { git = "https://github.com/golemfactory/ya-client.git", rev = "4099adb99dadbc8929caba33538806f5e10d5920"}
+ya-client = { git = "https://github.com/golemfactory/ya-client.git", rev = "c3f3419e48eade7f4d491c71948a1a50446d87d5"}
```
Comment (tworec) [inline Cargo.toml:177]: pls publish `ya-client` as 0.5.1 to crates.io and remove this patch before beta.1

Comment (maaktweluit) [reply Cargo.toml:177]: i have no idea how :) is there a guide for this?

File: core/payment-driver/zksync/src/zksync/wallet.rs (hunk 4 above)
Comment (maaktweluit) [reply core/payment-driver/zksync/src/zksync/wallet.rs:289]: as i said i made it a better message, seems the update got missing... checking

Comment (maaktweluit) [reply core/payment-driver/zksync/src/zksync/wallet.rs:289]: seems all info is printed as needed, so removed the ugly line on `payments/beta-1` branch

File: core/payment-driver/zksync/src/zksync/wallet.rs (hunk 7 above)
Comment (maaktweluit) [reply core/payment-driver/zksync/src/zksync/wallet.rs:353]: in an upcoming PR i'm cleaning all the GNT names, will fix this there. ( to master )

File: Cargo.toml (hunk 8 above)
Comment (tworec) [reply Cargo.toml:177]: done! :)

**Pull request merged**